test/node/preprocess/encoder.ts

```diff
@@ -0,0 +1,62 @@
+import { eigenSolve } from '../../../src/linalg/eigen';
+import '@tensorflow/tfjs-backend-cpu';
+import * as tf from '@tensorflow/tfjs-core';
+import { tensorEqual } from '../../../src/linalg/utils';
+import { assert } from 'chai';
+import 'mocha';
+import { OneHotEncoder } from '../../../src/preprocess';
+
+const x = ['tree', 'apple', 'banana', 'tree', 'apple', 'banana'];
+const xEncode = tf.tensor([
+  [1, 0, 0],
+  [0, 1, 0],
+  [0, 0, 1],
+  [1, 0, 0],
+  [0, 1, 0],
+  [0, 0, 1],
+]);
+const xEncodeDrop = tf.tensor([
+  [0, 0],
+  [1, 0],
+  [0, 1],
+  [0, 0],
+  [1, 0],
+  [0, 1],
+]);
+const bx = ['tree', 'apple', 'tree', 'apple'];
+const bxEncodeDrop = tf.tensor([0, 1, 0, 1]);
+
+describe('OneHot Encoder', () => {
+  it('encode', async () => {
+    const encoder = new OneHotEncoder();
+    await encoder.init(x);
+    const xOneHot = await encoder.encode(x);
+    assert.deepEqual(xOneHot.dataSync(), xEncode.dataSync());
+  });
+  it('decode', async () => {
+    const encoder = new OneHotEncoder();
+    await encoder.init(x);
+    const xCate = await encoder.decode(xEncode);
+    // @ts-ignore
+    assert.deepEqual(xCate.dataSync(), x);
+  });
+  it('encode drop first', async () => {
+    const encoder = new OneHotEncoder({drop: 'first'});
+    await encoder.init(x);
+    const xOneHot = await encoder.encode(x);
+    assert.deepEqual(xOneHot.dataSync(), xEncodeDrop.dataSync());
+  });
+  it('encode binary only', async () => {
+    const encoder = new OneHotEncoder({drop: 'binary-only'});
+    await encoder.init(bx);
+    const bxOneHot = await encoder.encode(bx);
+    assert.deepEqual(bxOneHot.dataSync(), bxEncodeDrop.dataSync());
+  });
+  it('decode binary only', async () => {
+    const encoder = new OneHotEncoder({drop: 'binary-only'});
+    await encoder.init(bx);
+    const bxCate = await encoder.decode(bxEncodeDrop);
+    // @ts-ignore
```

Comment:
Why ignore the following line?

Reply:
use dataSync() function on a string tensor will cause type error, the return types of dataSync() are  limited to numeric arrays

Reply:
Use the keyword `as` to declare the mentioned case instead of `ts-ignore` which is lack of details.